### PR TITLE
server/storage/mvcc: fix oss-fuzz issue 44449

### PR DESCRIPTION
**Pull request by AdamKorcz**
Fixes: https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=44449&q=etcd&can=1

### CODE REVIEW

File: server/storage/mvcc/index.go [hunk 1]
```diff
@@ -257,6 +257,9 @@ func (ti *treeIndex) Equal(bi index) bool {
 	equal := true
 
 	ti.tree.Ascend(func(item btree.Item) bool {
+		if _, ok := b.tree.Get(item).(*keyIndex); !ok {
+			return false
+		}
 		aki := item.(*keyIndex)
 		bki := b.tree.Get(item).(*keyIndex)
```
Comment (serathius) [inline server/storage/mvcc/index.go:264]:
```suggestion
		if bki, ok := b.tree.Get(item).(*keyIndex); !ok {
			return false
		}
		aki := item.(*keyIndex)
```
Should we also check `aki`?

**Review by serathius — APPROVED** [review]
LGTM, cc @ahrtr @ptabor @spzala

File: server/storage/mvcc/index.go [hunk 2]
```diff
@@ -257,8 +257,14 @@ func (ti *treeIndex) Equal(bi index) bool {
 	equal := true
 
 	ti.tree.Ascend(func(item btree.Item) bool {
-		aki := item.(*keyIndex)
-		bki := b.tree.Get(item).(*keyIndex)
+		var aki, bki *keyIndex
+		var ok bool
```
Comment (ahrtr) [inline server/storage/mvcc/index.go:261]: nit, use form something like below,
```
var (
    aki, bki  *keyIndex
    ok bool
)
```

**Review by spzala — APPROVED** [review]
lgtm
Thanks @AdamKorcz

**Pull request merged**